src/loaders/tvg/tvgTvgLoader.cpp

```diff
@@ -91,6 +91,9 @@ bool TvgLoader::readHeader()
 
     ptr += TVG_HEADER_COMPRESS_SIZE;
 
+    //Make sure the pointer does not exceed the size now
+    if (ptr > this->data + this->size) return false;
```

Comment:
I think we can return immediately in the beginning of readHeader()
if (size < TVG_HEADER_SIZE) return false;

if the wrong size is given, it can be less than 1.

Reply:
ok done